src/moveit_opw_kinematics_plugin.cpp

```diff
@@ -497,34 +497,43 @@ bool MoveItOPWKinematicsPlugin::setOPWParameters()
 
   ros::NodeHandle nh;
 
-  std::map<std::string, double> geometric_parameters, dummy;
-  if (!lookupParam("opw_kinematics_geometric_parameters", geometric_parameters, dummy))
+  std::map<std::string, double> geometric_parameters;
+  if (!lookupParam("opw_kinematics_geometric_parameters", geometric_parameters, {}))
   {
     ROS_ERROR_STREAM("Failed to load geometric parameters for ik solver.");
     return false;
   }
 
-  std::vector<double> joint_offsets, dummy2;
-  if (!lookupParam("opw_kinematics_joint_offsets", joint_offsets, dummy2))
+  std::vector<double> joint_offsets;
+  if (!lookupParam("opw_kinematics_joint_offsets", joint_offsets, {}))
   {
     ROS_ERROR_STREAM("Failed to load joint offsets for ik solver.");
     return false;
   }
 
-  std::vector<int> joint_sign_corrections, dummy3;
-  if (!lookupParam("opw_kinematics_joint_sign_corrections", joint_sign_corrections, dummy3))
+  std::vector<int> joint_sign_corrections;
+  if (!lookupParam("opw_kinematics_joint_sign_corrections", joint_sign_corrections, {}))
   {
     ROS_ERROR_STREAM("Failed to load joint sign corrections for ik solver.");
     return false;
   }
 
   opw_parameters_.a1 = geometric_parameters["a1"];
   opw_parameters_.a2 = geometric_parameters["a2"];
-  opw_parameters_.b = geometric_parameters["b"];
+  opw_parameters_.b  = geometric_parameters["b"];
   opw_parameters_.c1 = geometric_parameters["c1"];
   opw_parameters_.c2 = geometric_parameters["c2"];
   opw_parameters_.c3 = geometric_parameters["c3"];
   opw_parameters_.c4 = geometric_parameters["c4"];
+
+  if (joint_offsets.size() != joint_sign_corrections.size())
```

Comment:
I wonder if it is even better to check if they have both a length of 6? Any other length would be invalid, given that opw_parameters are specific to 6 dof robots.

Reply:
Yes, that would be better.

I wasn't sure any more whether it's *always* six. It's been a while since I looked at the paper.

Reply:
I guess the algorithm can also be used for robots with less than 6 dofs. But `opw_kinematics` does not support this. If the size is larger than 6, the following line would be a problem:
```C++
opw_parameters_.offsets[i] = joint_offsets[i];
```
as offsets a fixed size array of length 6.

The other case I considered is adding support for redundant dofs, but this would not change the opw_parameters, only the implementation of the plugin.

Reply:
I'll see if I can just check for `size() == 6`.

If you have a chance to add it, feel free. My fork is not on an organisational account, so you should be able to push commits to it.

Reply:
I added it. I think it is ready to be merged now.